runtime/src/identity.rs

```diff
@@ -399,6 +415,42 @@ decl_module! {
             Ok(())
         }
 
+        fn forwarded_call(origin, target_did: IdentityId, proposal: Box<T::Proposal>) -> Result {
+            let sender = ensure_signed(origin)?;
+
+            // 1. Constraints.
+            // 1.1. A valid current identity.
+            if let Some(current_did) = <CurrentDid>::get() {
+                // 1.2. Check that current_did is a signing key of target_did
+                ensure!( Self::is_authorized_identity(current_did, target_did),
+                    "Current identity cannot be forwarded, it is not a signing key of target identity");
+            } else {
+                return Err("Missing current identity on the transaction");
+            }
+
+            // 1.3. Check that target_did has a KYC (this has
+            // already been done for current_did in the SignedExtension)
+            ensure!(Self::has_valid_kyc(target_did), "Invalid KYC validation on target did");
```

Comment:
If someone does nested `forwarded_call` then the first current_did will be checked by `SignedExtension` but the current_did of the nested forwarded call will not be. Forwarding call bypasses signed extensions. However, this is not an issue for us because the current_did of nested forwarded call was target did of first forwarded call and got verified in that way. Perhaps we should just edit/remove the code comment.

Reply:
Make sense.